src/services/swap_service.ts

```diff
@@ -336,18 +365,35 @@ export class SwapService {
             excludedSources,
             apiKey,
             rfqt,
+            swapVersion,
             // tslint:disable-next-line:boolean-naming
         } = params;
         let _rfqt;
         if (apiKey !== undefined && (isETHSell || from !== undefined)) {
+            let takerAddress;
+            switch (swapVersion) {
+                case SwapVersion.V0:
+                    // If this is a forwarder transaction, then we want to request quotes with the taker as the
+                    // forwarder contract. If it's not, then we want to request quotes with the taker set to the
+                    // API's takerAddress query parameter, which in this context is known as `from`.
+                    takerAddress = isETHSell ? this._contractAddresses.forwarder : from || '';
+                    break;
+                case SwapVersion.V1:
+                    // If this is an ETH sell, then we want to request quotes with the taker address
+                    // as the ExchangeProxy so that it can automatically unwrap WETH to ETH. If it's not,
+                    // then we want to request quotes with the taker set to the API's takerAddress query
+                    // parameter, which in this context is known as `from`.
+                    takerAddress = isETHSell ? await this._getExchangeProxyFlashWalletAsync() : from || '';
```

Comment:
This is ALWAYS the ExchangeProxy flash wallet in the new architecture. It always is the taker of orders (both in ETH sells and DAI sells)